addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ExtensionGraphQl.java

```diff
@@ -141,6 +148,33 @@ protected GraphQlParam getParam() {
         return param;
     }
 
+    public void addParserThread(ParserThread thread) {
+        parserThreads.add(thread);
+    }
+
+    private void stopParserThreads() {
```

Comment:
We should clear the list too, to not keep references to the threads longer than needed (ideally the thread/parser would remove itself once done too).